tests/test_format_detection.py

```diff
@@ -0,0 +1,169 @@
+from unittest import TestCase
+import os.path as osp
+
+from datumaro.components.format_detection import (
+    FormatDetectionConfidence, FormatRequirementsUnmet, apply_format_detector,
+)
+from datumaro.util.test_utils import TestDir
+
+from tests.requirements import Requirements, mark_requirement
+
+
+class FormatDetectionTest(TestCase):
+    def setUp(self) -> None:
+        test_dir_context = TestDir()
+        self._dataset_root = test_dir_context.__enter__()
+        self.addCleanup(test_dir_context.__exit__, None, None, None)
+
+    @mark_requirement(Requirements.DATUM_GENERAL_REQ)
+    def test_empty_detector(self):
+        result = apply_format_detector(self._dataset_root, lambda c: None)
+        self.assertEqual(result, FormatDetectionConfidence.MEDIUM)
```

Comment:
Do you think we need to check if context was used at all?

Reply:
Are you suggesting to add this check to `apply_format_detector`?

I don't think that would be useful in practice. A detector that does not use the context will result in a detection on any directory, which is weird, but not really invalid. If such a detector is mistakenly added to one of the built-in plugins, that will be caught by existing tests (since the existing formats will no longer be unambiguous). And if a user writes a custom plugin with an empty detector, well... why not let them? Maybe they don't want to bother writing a real detector because they are only going to work with their custom format anyway.

Reply:
I not insist on having such check, but I'm trying to guess possible reasons to allow such situations. If a user don't want to implement a detector, he just doesn't define the class method. If has empty implementation, it is unlikely to be working in practice. However, probably, it can be useful in some situations, where the user just don't want to use the context and just returns confidence.

Reply:
> I not insist on having such check, but I'm trying to guess possible reasons to allow such situations.

FWIW, my main reasons are:
1. It doesn't break the contract.
2. It would take extra work to forbid it, and the benefit seems miniscule.